### PR TITLE
(#5591) - remove extras API (BREAKING CHANGE)

### PR DESCRIPTION
**Pull request by nolanlawson**
Note that this doesn't remove the `dist` plugins (`pouchdb.memory.js` etc.). Instead, this just makes it so that `require('pouchdb/extras/*')` no longer works.

Removing the `dist` plugins may be a longer-term effort and might not even be feasible in 6.0.0. I'm opening a new issue: #5592.

### CODE REVIEW

**Comment by nolanlawson** [comment]
Style problems, should be fixed now...

**Comment by daleharvey** [comment]
Very happy to remove these, 👍 when green

**Comment by gr2m** [comment]
that looks like  a proper bug I think? https://travis-ci.org/pouchdb/pouchdb/jobs/154895463#L1594

And I guess the `COMMAND=test-extras` can be removed from the test matrix with this PR?

**Comment by nolanlawson** [comment]
That is probably a bug in the test itself. I'll take a look when I have a moment.

**Comment by gr2m** [comment]
I’m looking into it right now

**Comment by gr2m** [comment]
I pushed to commits that should fix the two failing builds

**Comment by daleharvey** [comment]
Awesome thanks @gr2m, small nit in future if you could avoid the style changes that arent related, our general style would lean towards the original `}, {` however the performance tests need quite a bit of cleaning up anyway so will get addresssed later.

File: tests/performance/index.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ if (startNow) {
   if (!process.browser) {
     // the two strings are to fool Browserify, because this test
     // fails in Node 0.11-0.12 due to sqlite3 being incompatible
-    global.PouchDB.plugin(require('../../packages/node_modules/' +
+    PouchDB.plugin(require('../../packages/node_modules/' +
```
Comment (nolanlawson) [inline tests/performance/index.js]: Not sure what's going on here, but LGTM!

**Comment by nolanlawson** [comment]
Thanks a lot @gr2m, been swamped and needed some help here! ❤️

File: tests/performance/index.js (hunk 1 above)
Comment (gr2m) [inline tests/performance/index.js]: I thought it might be something that you missed before in the cleanup, I couldn’t find it out myself

**Comment by gr2m** [comment]
You are more than welcome :) We are really looking forward to the 6.0.0 release so that’s the least I can do

**Pull request merged**